### PR TITLE
Add a warning dialog before merging

### LINKED ISSUE #3816: Warning dialog before merging
Add a warning dialog when clicking on merge before proceeding with merge. 
*See @grantfitzsimmons text

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixes #3816

### CODE REVIEW

**Review by carlosmbe — COMMENTED** [review]
It works fine. I'd approve it but just want to ask something.

Is it possible for us to make the warning multiple lines. Have it stretch across the entire screen feels a bit more eye straining compared to have 3 lines in the middle of the screen. If not eye straining, then a slight inconvenience.

Although, I suppose that extra effort does make sure we're communicating the gravity of situation. 


Screenshot 2023-07-20 at 4 26 31 PM

## VS

Just a rough mock up. Once more this is just my personal opinion.

Screenshot 2023-07-20 at 4 37 32 PM

**Comment by maxxxxxdlp** [comment]
> Is it possible for us ...

Yes. @CarolineDenis you can use the `className={}` prop on dialog for that

File: specifyweb/frontend/js_src/lib/components/Merging/index.tsx
Comment (maxxxxxdlp) [inline specifyweb/frontend/js_src/lib/components/Merging/index.tsx:282]: can simplify the logic and improve UX:
as soon as the user dismisses the warning, trigger the form submit - save user a click and removes the need for isAccepted state

File: specifyweb/frontend/js_src/lib/components/Merging/index.tsx
Comment (maxxxxxdlp) [inline specifyweb/frontend/js_src/lib/components/Merging/index.tsx:296]: was the change from `Submit` to `Button` intentional? should be ok, just want to double check

**Review by maxxxxxdlp — CHANGES_REQUESTED** [review]
Blocking this PR until https://github.com/specify/specify7/issues/3816#issuecomment-1644896154 is answered as that's quite important

**Review by bronwyncombs — CHANGES_REQUESTED** [review]
We should change the way the warning dialog is remembered, or possibly just how it is related to the merge records dialog. 

Currently, after the warning dialog is moved, the merge record dialog will be that size the next open. Further, the warning dialog will then conform to the size of the agent merge dialog despite changing the agent merge dialog before prompting the warning. 

We could also opt to disable the ability to change the warning dialog size altogether if that's a more timely option. 
(As per @carlosmbe comment above, we'd have to discuss what that size would be.)

https://github.com/specify/specify7/assets/135047322/a4b6184f-9395-4a33-a49e-d40c2a899344

File: specifyweb/frontend/js_src/lib/components/Merging/index.tsx
Comment (CarolineDenis) [reply specifyweb/frontend/js_src/lib/components/Merging/index.tsx:296]: yes but can revert is it's better to have submit.red

**Review by grantfitzsimmons — CHANGES_REQUESTED** [review]
We should add a checkbox as @maxxxxxdlp mentioned if it is not too difficult that says
 "Don't remind me again" to the left of the **Proceed** button. Will this add too much complexity?

**Review by grantfitzsimmons — CHANGES_REQUESTED** [review]
Could we replace the **Proceed** with **Merge** again?

image

Second, could we put the checkbox to the left of the "Accept" button?
Third, could we rename the "Accept" button to "Proceed"?

image

Fourth and finally, could we have the "Merge Records" warning dialog be smaller and centered instead of very wide? Those are all my comments!

Looking good!

**Review by carlosmbe — COMMENTED** [review]
Looks good to me. Can confirm the new layout and Do Not Show Again Button work.

**Comment by CarolineDenis** [comment]
NOTE: to have the check box  'Don't show this again' back => go to user-tools and clear cache

**Comment by maxxxxxdlp** [comment]
@CarolineDenis if you create a new dataset without uploading (using the New button in the data sets window) you will see a dialog, that includes such checkbox (with local storage memory)

**Review by grantfitzsimmons — CHANGES_REQUESTED** [review]
image

Looks good! One weird thing before we merge– you can't uncheck this box once it is checked?

Firefox, macOS Sonoma

File: specifyweb/frontend/js_src/lib/components/Merging/index.tsx
Comment (maxxxxxdlp) [inline specifyweb/frontend/js_src/lib/components/Merging/index.tsx:266]: (optional) rename `toggleWarningDialog` to `handleToggleWarning` for consistency with how handlers are usually called?

File: specifyweb/frontend/js_src/lib/components/Merging/index.tsx
Comment (maxxxxxdlp) [inline specifyweb/frontend/js_src/lib/components/Merging/index.tsx:345]: no need for `${` and `}` - just use the variable without backticks

File: specifyweb/frontend/js_src/lib/components/Merging/index.tsx
Comment (maxxxxxdlp) [inline specifyweb/frontend/js_src/lib/components/Merging/index.tsx:348]: no need for `<>` and `</>`

**Review by maxxxxxdlp — APPROVED** [review]
only minor things, besides the bug Grant mentioned

**Review by maxxxxxdlp — APPROVED** [review]

**Review by bronwyncombs — APPROVED** [review]
Looks good! Checkbox working and cache behaves as expected.

https://github.com/specify/specify7/assets/135047322/f18fab6b-f114-4f3b-addc-8d21330283e5

The dialog is constrained horizontally but the size can still be adjusted vertically. Re-sizing in this way is remembered, but not by the merge records dialog. 

https://github.com/specify/specify7/assets/135047322/d728c62d-5c55-4cc7-9d4b-829ac1fd38c9

**Pull request merged**
